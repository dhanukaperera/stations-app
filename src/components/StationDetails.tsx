import ReactAudioPlayer from 'react-audio-player';
import styled from 'styled-components';
import { ALT_LOGO } from '../constants/constants';

type StationDetailsProps = {
	name: string
	tagline: string,
	brandLogo: string,
	steamingUrl: string
}

const StationDetails = (props: StationDetailsProps) => {
	const { name, tagline, brandLogo, steamingUrl } = props;

	return <StationDetailsStyles>
		<img src={brandLogo} alt={`${name} ${ALT_LOGO}`} />
		<InfoContainer>
			<h1>
				{name}
			</h1>
			<p>{tagline}</p>
			<ReactAudioPlayer
				src={steamingUrl}
				controls
			/>
		</InfoContainer>
	</StationDetailsStyles>
}
export default StationDetails

const StationDetailsStyles = styled.div`
	display:flex;
	align-items:center;
	padding:4rem 1rem;
	margin:2rem;
	img {
		border:4px solid #006efa;
		border-radius: 50%;
	}
	box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.2);
`
const InfoContainer = styled.div`
	margin:1rem 2rem;
	h1  {
		margin-bottom:0.5rem;
	}
	p{
		margin-bottom:1rem;
	}
`