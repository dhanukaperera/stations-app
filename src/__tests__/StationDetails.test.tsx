import { expect, test } from 'vitest'
import { render } from '@testing-library/react'
import StationDetails from '../components/StationDetails'
import { StaticRouter } from 'react-router-dom/server';


test("it should render a station details", () => {
	const { asFragment } = render(
		<StaticRouter location={"/"} >
			<StationDetails
				name='Capital Teesside'
				tagline="The UK's No.1 Hit Music Station"
				brandLogo='https://herald.musicradio.com/media/0d3d891d-32f2-4c53-95ee-8d1e35bdd126.png'
				steamingUrl='https://media-ssl.musicradio.com/CapitalTeesside' />
		</StaticRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});