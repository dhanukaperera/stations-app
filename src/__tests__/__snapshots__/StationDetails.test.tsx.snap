// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it should render a list of  1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB dzAcBK"
  >
    <img
      alt="Capital Teesside logo"
      src="https://herald.musicradio.com/media/0d3d891d-32f2-4c53-95ee-8d1e35bdd126.png"
    />
    <div
      class="sc-gueYoa kEobtL"
    >
      <h1>
        Capital Teesside
      </h1>
      <p>
        The UK's No.1 Hit Music Station
      </p>
      <audio
        class="react-audio-player "
        controls=""
        id=""
        preload="metadata"
        src="https://media-ssl.musicradio.com/CapitalTeesside"
        title="https://media-ssl.musicradio.com/CapitalTeesside"
      >
        <p>
          Your browser does not support the 
          <code>
            audio
          </code>
           element.
        </p>
      </audio>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it should render a station deatils 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB dzAcBK"
  >
    <img
      alt="Capital Teesside logo"
      src="https://herald.musicradio.com/media/0d3d891d-32f2-4c53-95ee-8d1e35bdd126.png"
    />
    <div
      class="sc-gueYoa kEobtL"
    >
      <h1>
        Capital Teesside
      </h1>
      <p>
        The UK's No.1 Hit Music Station
      </p>
      <audio
        class="react-audio-player "
        controls=""
        id=""
        preload="metadata"
        src="https://media-ssl.musicradio.com/CapitalTeesside"
        title="https://media-ssl.musicradio.com/CapitalTeesside"
      >
        <p>
          Your browser does not support the 
          <code>
            audio
          </code>
           element.
        </p>
      </audio>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it should render a station details 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB dzAcBK"
  >
    <img
      alt="Capital Teesside logo"
      src="https://herald.musicradio.com/media/0d3d891d-32f2-4c53-95ee-8d1e35bdd126.png"
    />
    <div
      class="sc-gueYoa kEobtL"
    >
      <h1>
        Capital Teesside
      </h1>
      <p>
        The UK's No.1 Hit Music Station
      </p>
      <audio
        class="react-audio-player "
        controls=""
        id=""
        preload="metadata"
        src="https://media-ssl.musicradio.com/CapitalTeesside"
        title="https://media-ssl.musicradio.com/CapitalTeesside"
      >
        <p>
          Your browser does not support the 
          <code>
            audio
          </code>
           element.
        </p>
      </audio>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders correctly with some stations 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB dzAcBK"
  >
    <img
      alt="Capital Teesside logo"
      src="https://herald.musicradio.com/media/0d3d891d-32f2-4c53-95ee-8d1e35bdd126.png"
    />
    <div
      class="sc-gueYoa kEobtL"
    >
      <h1>
        Capital Teesside
      </h1>
      <p>
        The UK's No.1 Hit Music Station
      </p>
      <audio
        class="react-audio-player "
        controls=""
        id=""
        preload="metadata"
        src="https://media-ssl.musicradio.com/CapitalTeesside"
        title="https://media-ssl.musicradio.com/CapitalTeesside"
      >
        <p>
          Your browser does not support the 
          <code>
            audio
          </code>
           element.
        </p>
      </audio>
    </div>
  </div>
</DocumentFragment>
`;
